lib/Service/PointService.php

```diff
@@ -952,19 +952,65 @@ private function orderByDate(array $points, array $dates): array {
 	 *
 	 */
 	public function purgeRestoringPoints(): void {
-		$c = $this->configService->getAppValue(ConfigService::STORE_ITEMS);
-		$i = 0;
-		foreach ($this->getLocalRestoringPoints(0, 0, false) as $point) {
-			if ($point->isArchive()) {
-				continue;
```

Comment:
(A) Archive-flag is no longer taken into account here when deleted.
So also restoring points are deleted which shall be kept infinitely.

Reply:
I couldn't figure out what an "Archive" in the backups was so I figured it was dead code. I'd be happy to incorporate it if you could point at some documentation on how it's supposed to work or explain it.

Reply:
I also didn't find any documentation about that. But I had a closer look at the `occ backup` commands output and found the commands `occ backup:point:archive` and `occ backup:point:unarchive`.

When using the archive command, the corresponding restoring point will be marked with an 'A':
| **A** 20220306151749-full-wLICFgahr9V6Nio         | 2022-03-06 15:17:49 | ...
And when using the unarchive command, the 'A' will be removed.

After next purge the local points marked with 'A' were still available and all others were deleted.

Reply:
Hrmm... that seems like an problematic function but I guess I see the logic. That should be an easy fix I'll graft in here as soon as I get some thinking time.